scheduler/src/main/java/io/camunda/zeebe/scheduler/future/CompletableActorFuture.java

```diff
@@ -48,6 +50,10 @@
   private Throwable failureCause;
   private final ManyToOneConcurrentLinkedQueue<ActorTask> blockedTasks =
       new ManyToOneConcurrentLinkedQueue<>();
+
+  private final ManyToOneConcurrentLinkedQueue<BiConsumer<V, Throwable>> blockedCallbacks =
```

Comment:
:wrench: `ManyToOneConcurrentLinkedQueue` is pretty heavy and by adding this queue here, all our futures have to allocate this.
Could we either find a common interface so that we can merge `blockedCallbacks` and `blockedTasks` queues or lazy-initialize `blockedCallbacks`?
Tbh I'm not sure how much difference it really makes but I think it makes sense to try and keep our actor classes lightweight.

Reply:
Why do you think `ManyToOneConcurrentLinkedQueue` is heavy?

Reply:
Using [jol](https://openjdk.org/projects/code-tools/jol/) to measure the size of a newly created instance of `CompletableActorFuture` on this branch:
```
io.camunda.zeebe.scheduler.future.CompletableActorFuture@24569dbad footprint:
     COUNT       AVG       SUM   DESCRIPTION
         1        56        56   io.camunda.zeebe.scheduler.future.CompletableActorFuture
         1        24        24   java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject
         1        16        16   java.util.concurrent.locks.ReentrantLock
         1        32        32   java.util.concurrent.locks.ReentrantLock$NonfairSync
         2       216       432   org.agrona.concurrent.ManyToOneConcurrentLinkedQueue
         2        24        48   org.agrona.concurrent.ManyToOneConcurrentLinkedQueuePadding1$Node
         8                 608   (total)
```

vs main:
```
io.camunda.zeebe.scheduler.future.CompletableActorFuture@24569dbad footprint:
     COUNT       AVG       SUM   DESCRIPTION
         1        48        48   io.camunda.zeebe.scheduler.future.CompletableActorFuture
         1        24        24   java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject
         1        16        16   java.util.concurrent.locks.ReentrantLock
         1        32        32   java.util.concurrent.locks.ReentrantLock$NonfairSync
         1       216       216   org.agrona.concurrent.ManyToOneConcurrentLinkedQueue
         1        24        24   org.agrona.concurrent.ManyToOneConcurrentLinkedQueuePadding1$Node
         6                 360   (total)
```

So the size increased from 360 to 608 bytes. In total that's less than expected but still shows that the size increases by 1.68x.